### DISCUSSION TITLE
Ability to customize what happens after record is created/edited.

### DISCUSSION THREAD

**Discussion by MrJoy** [avo-hq/avo#776] [Feature requests]
In some workflows, it makes more sense to redirect the user to, say, the index view after creating/editing a record than to send the user to the show view.  Having the ability to gracefully customize this behavior would be quite helpful.

I've reviewed the Avo code and it looks like changing this behavior right now would basically involve copy-pasting the `create` method into my resource class, and changing it there.  That's certainly a feasible short-term workaround but I'm concerned about drift, as the upstream code evolves.

**Comment by adrianthedev** [comment]
Hi,

That's not such a bad idea. I'm thinking about adding the following methods `after_save_path` and `after_update_path` like so

```ruby
# redirect the user to the index page
self.after_save_path =  :index

# redirect to show (default)
self.after_save_path =  :show
```

Do you think that would be enough?

**Reply by MrJoy** [reply]
It's certainly an excellent start, and would probably cover 80% of user needs.

I could certainly imagine situations where it _wouldn't_ be enough but I don't think you should hold off on implementing what you've suggested in favor of sorting out a "perfect" solution.

**Reply by MrJoy** [reply]
Rather, "80% of user needs, including my immediate need".

**Reply by adrianthedev** [reply]
I agree! Thanks for the feedback 👍 

On the [roadmap](https://github.com/orgs/avo-hq/projects/3/views/1) to take on next.

**Reply by adrianthedev** [reply]
At some point we added [controller hooks](https://docs.avohq.io/3.0/controllers.html) too.
